src/tribler/gui/widgets/downloadspage.py

```diff
@@ -119,45 +115,43 @@ def start_loading_downloads(self):
         self.window().downloads_list.addTopLevelItem(self.loading_message_widget)
         self.window().downloads_list.setItemWidget(self.loading_message_widget, 2, self.loading_list_item)
 
-    def schedule_downloads_timer(self, now=False):
-        interval_msec = REFRESH_DOWNLOADS_NOW_INTERVAL_MSEC if now else REFRESH_DOWNLOADS_INTERVAL_MSEC
-        self.downloads_timer.start(interval_msec)
+    def schedule_downloads_refresh(self, interval_msec=REFRESH_DOWNLOADS_BACKGROUND_INTERVAL_MSEC):
+        remaining = self.background_refresh_downloads_timer.remainingTime()
+        if remaining > 0:
+            interval_msec = min(remaining, interval_msec)
+        self.background_refresh_downloads_timer.start(interval_msec)
 
-        # we also start a timer to reschedule the `downloads_timer` after timed out.
-        self.downloads_timeout_timer.start(interval_msec + WAITING_TIME_BEFORE_THE_TIMEOUT_MSEC)
+    def on_background_refresh_downloads_timer(self):
+        self.refresh_downloads()
+        self.schedule_downloads_refresh()
 
-    def on_downloads_request_timeout(self):
+    def background_refresh_downloads(self):
         self._logger.warning("Downloads request timed out, retrying...")
         if self.rest_request:
             self.rest_request.cancel()
-        self.schedule_downloads_timer()
+        self.schedule_downloads_refresh()
+
+    def stop_refreshing_downloads(self):
+        self.background_refresh_downloads_timer.stop()
+
+    def refresh_downloads(self):
+        if self.rest_request:
+            self.rest_request.cancel()
```

Comment:
> I have a concern regarding this part of the code. Before this PR, request canceling was managed by a separate timer. Although the previous logic with two separate timers wasn't very elegant, it did ensure that a new request would not cancel an ongoing request that the Core was still processing.

Unfortunately, I disagree with this sentence. The previous cancellation logic works roughly the same way that it works in the PR. 

A request could be canceled in two places in the previous version of the code:

1. By calling `on_downloads_request_timeout` (removed in this PR)

```python
    def on_downloads_request_timeout(self):
        self._logger.warning("Downloads request timed out, retrying...")
        if self.rest_request:
            self.rest_request.cancel()
        self.schedule_downloads_timer()
```

2. On each `load_downloads` call (renamed with `refresh_downloads`)
```python
    def load_downloads(self):
        url_params = {'get_pieces': 1}
        if self.window().download_details_widget.currentIndex() == 3:
            url_params['get_peers'] = 1
        elif self.window().download_details_widget.currentIndex() == 1:
            url_params['get_files'] = 1

        isactive = not self.isHidden()

        if isactive or (time.time() - self.downloads_last_update > 30):
            # Update if the downloads page is visible or if we haven't updated for longer than 30 seconds
            self.downloads_last_update = time.time()
            priority = QNetworkRequest.LowPriority if not isactive else QNetworkRequest.HighPriority
            if self.rest_request:
                self.rest_request.cancel()

            request_manager.get(
                endpoint="downloads",
                url_params=url_params,
                on_success=self.on_received_downloads,
                priority=priority,
            )
```

Therefore additional timer has nothing in common with "ensuring that a new request would not cancel an ongoing request".

But the fanny fact is... it seems that `self.rest_request` is always equal to `None`, so, cancellation logic could be safely removed without changing an application logic.

Reply:
The logic with the removed `self.rest_request` value assignment was introduced here: https://github.com/Tribler/tribler/commit/f5d6aeb2e4befc55c1385ce6596393b95aa4d7c1#diff-d512cc9b333c64acd7568dfce806cc67707f68e9ff9016368a096a66921a353f

Reply:
> The logic with the removed self.rest_request value assignment was introduced here

Ah, this is the reason why we stopped seeing "Request was canceled" spam for the download endpoint :)

I restored the assignment, and now the log is again full with
```
Request was canceled: GET http://localhost:52194/downloads?get_pieces=1
```

Under normal circumstances, frequent cancellation of download requests shouldn't be required. So, there's still a need for a solution to address this issue.

Reply:
Turns out, the problem with request canceling has a simple answer: if we assign `self.rest_request`, we also need to clean it in `on_received_downloads` :) With this change, the `Request was cancelled` spam is gone, so it turns out to be a false positive.

After this change, I measured the response time from the `/downloads` endpoint on my machine, and it is usually 0.3 - 0.5 seconds but periodically 3.7 - 4.0 seconds. Only these two intervals can be observed, nothing in-between, like "1.0 second". It is possible that the endpoint logic sometimes gets blocked by other asynchronous processes, and this causes a bigger delay.

With this, I think it may be too early to cancel a request if we issue a new request just tens of milliseconds after the previous one (in case of a request provoked by UI changes). We need to have some timeout, not necessarily too long, say, 10 seconds, and not cancel the previous request before the timeout.

Reply:
Together with @kozlovsky, we decided on the following:

1. In this PR, I will remove the cancellation logic.
2. In the subsequent PR, I will write code to ensure the order of responses.